lib/miq_automation_engine/engine/miq_ae_service/miq_ae_service_rbac.rb

```diff
@@ -0,0 +1,54 @@
+module MiqAeMethodService
+  module MiqAeServiceRbac
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      def find_ar_object_by_id(id)
+        if rbac_enabled?
+          Rbac.filtered(model.where(:id => id), :user => workspace.ae_user).first
+        else
+          model.find(*id)
+        end
+      end
+
+      def all
+        objs = rbac_enabled? ? Rbac.filtered(model, :user => workspace.ae_user) : model.all
+        wrap_results(objs)
+      end
+
+      def count
+        rbac_enabled? ? Rbac.filtered(model, :user => workspace.ae_user).count : model.count
+      end
+
+      def first
+        objs = rbac_enabled? ? Rbac.filtered(model, :user => workspace.ae_user, :limit => 1) : model
+        wrap_results(objs.first)
+      end
+
+      def filter_objects(objs)
+        if objs.nil?
+          objs
+        elsif objs.kind_of?(Array) || objs.kind_of?(ActiveRecord::Relation)
+          rbac_enabled? ? Rbac.filtered(objs, :user => workspace.ae_user) : objs
+        else
+          rbac_enabled? ? Rbac.filtered_object(objs, :user => workspace.ae_user) : objs
+        end
+      end
+
+      def workspace
+        MiqAeEngine::MiqAeWorkspaceRuntime.current || MiqAeEngine::DrbRemoteInvoker.workspace
+      end
+
+      def workspace_from_drb_thread
```

Comment:
@gmcculloug 
This was the old method which used the module function front to get to the workspace. I will get rid of it